extensions/amp-script/0.1/amp-script.js

```diff
@@ -296,20 +290,29 @@ export class AmpScript extends AMP.BaseElement {
       container || this.element,
       workerAndAuthorScripts,
       config
-    ).then((workerDom) => {
-      this.workerDom_ = workerDom;
-      this.initialize_.resolve();
-      // workerDom will be null if it failed to init.
-      if (this.workerDom_) {
-        this.workerDom_.onerror = (errorEvent) => {
-          errorEvent.preventDefault();
-          user().error(
-            TAG,
-            `${errorEvent.message}\n    at (${errorEvent.filename}:${errorEvent.lineno})`
-          );
-        };
-      }
-    });
+    )
+      .then((workerDom) => {
+        this.workerDom_ = workerDom;
+        this.initialize_.resolve();
+        // workerDom will be null if it failed to init.
+        if (this.workerDom_) {
+          this.workerDom_.onerror = (errorEvent) => {
+            errorEvent.preventDefault();
+            user().error(
+              TAG,
+              `${errorEvent.message}\n    at (${errorEvent.filename}:${errorEvent.lineno})`
+            );
+          };
+        }
+      })
+      .then(
+        () => {
+          this.setReadyState(ReadyState.COMPLETE);
```

Comment:
Same story. We can just return `layoutCallback()` from `mountCallback` and then the migration  is a bit simpler - it just doesn't allow reloading w/o remounting which, I think, is totally ok for non-fully Bento components.